adatest/_test_tree.py

```diff
@@ -273,6 +273,22 @@ def topic(self, topic):
         ids = [id for id, test in self._tests.iterrows() if is_subtopic(topic, test.topic)]
         return self.loc[ids]
 
+    def topic_has_direct_tests(self, target_topic: str)-> bool:
+        """Check if a topic has direct tests."""
+        hdt_df = self._tests.apply(
+            lambda row: row['topic']==target_topic and row['label'] != 'topic_marker',
+            axis=1
+        )
+        return hdt_df.any()
+
+    def topic_has_subtopics(self, target_topic: str) -> bool:
+        """Check if a topic has subtopics."""
+        has_subtopics_df = self._tests.apply(
+            lambda row: row['topic']!=target_topic and is_subtopic(target_topic, row["topic"]),
+            axis=1
+        )
+        return has_subtopics_df.any()
+
```

Comment:
If we still have performance issues we could try vector AND operations here. But this seems good for now :)

Reply:
If we stick with a topic tree, I'd be more inclined to rewrite the internal datastructure to be a tree, and just use DataFrame for serialisation.